openwisp_firmware_upgrader/base/models.py

```diff
@@ -172,7 +174,10 @@ def get_build_directory(instance, filename):
 class AbstractFirmwareImage(TimeStampedEditableModel):
     build = models.ForeignKey(get_model_name('Build'), on_delete=models.CASCADE)
     file = PrivateFileField(
-        'File', upload_to=get_build_directory, max_file_size=app_settings.MAX_FILE_SIZE
+        'File',
+        upload_to=get_build_directory,
+        max_file_size=app_settings.MAX_FILE_SIZE,
+        storage=PrivateFileSystemStorage(base_url=FIRMWARE_IMAGE_BASEURL),
```

Comment:
@atb00ker do we need both `upload_to` and `storage` attribute here? 
AFAIK base_url and upload_to serves the same purpose.

Reply:
I think it's not clear in the docs, but both options are doing different things: https://github.com/edoburu/django-private-storage#usage
1. `upload_to`: handles the location of the file of the server.
2. `storage` object: handles the url part! :smile:

Reply:
Thanks for clearing